editor/selectors.js

```diff
@@ -913,9 +914,10 @@ export const getMostFrequentlyUsedBlocks = createSelector(
 		const { blockUsage } = state.preferences;
 		const orderedByUsage = keys( blockUsage ).sort( ( a, b ) => blockUsage[ b ] - blockUsage[ a ] );
 		// add in paragraph and image blocks if they're not already in the usage data
-		return [ ...orderedByUsage, ...without( [ 'core/paragraph', 'core/image' ], ...orderedByUsage ) ]
-			.slice( 0, MAX_FREQUENT_BLOCKS )
-			.map( blockType => getBlockType( blockType ) );
+		return compact(
```

Comment:
We execute: `blockType => getBlockType( blockType )` for all listed blocks now. Whad do you think about using `reduce` instead:
```js
return [ ...orderedByUsage, ...without( [ 'core/paragraph', 'core/image' ], ...orderedByUsage ) ].reduce(
    (  blockTypes, blockName ) => {
        if ( blockTypes.length >= MAX_FREQUENT_BLOCKS ) {
            return blockTypes;
        }
        const blockType = getBlockType( blockName );
        return blockType ? [ ...blockTypes, blockType ] : blockTypes;
    },
    []
);
```
It might need some polishing. I'm not convinced it's better seeing what I coded above :)

Reply:
Performance wise this is probably better, but I personally favor clarity over premature optimization. I like the "clarity" of the more "composition/functional" approach.

Reply:
Fine with me.